src/components/MoveAdder.ts

```diff
@@ -15,4 +15,247 @@ export class MoveAdder<TGameStartr extends FullScreenPokemon> extends GeneralCom
+                    killOnB: ["GeneralText"],
+                });
+                this.gameStarter.menuGrapher.addMenuList("Yes/No", {
+                    options: [
+                        {
+                            text: "YES",
+                            callback: () => this.acceptLearnMove(pokemon, move),
+                        },
+                        {
+                            text: "NO",
+                            callback: () => this.refuseLearnMove(pokemon, move),
+                        }],
+                });
+                this.gameStarter.menuGrapher.setActiveMenu("Yes/No");
+        });
+    }
+
+    /**
+     * Brings up the dialog for selecting a move to replace with a new move.
+     *
+     * @param pokemon   The pokemon whose moveset is being modified.
+     * @param move   The move that's going to be added into the moveset.
+     */
+    private acceptLearnMove(pokemon: IPokemon, move: IMove) {
+        this.gameStarter.menuGrapher.deleteMenu("Yes/No");
+        this.gameStarter.menuGrapher.createMenu("GeneralText");
+        this.gameStarter.menuGrapher.addMenuDialog(
+            "GeneralText",
+            [
+                [
+                    "Which move should be forgotten?",
+                ],
+            ],
+            (): void => {
+                const moves: IMove[] = pokemon.moves;
+
+                const newPos: IMenuSchemaPosition = {
+                    offset: {
+                        top: -80,
+                        bottom: 80,
+                    },
+                };
+                this.gameStarter.menuGrapher.createMenu("BattleFightList", {
+                    position: newPos,
+                    killOnB: ["GeneralText"],
+                });
+                this.gameStarter.menuGrapher.addMenuList("BattleFightList", {
+                    options: [
+                        {
+                            text: moves[0].title.toUpperCase(),
+                            callback: () => this.teachMove(pokemon, move, 0),
+                        },
+                        {
+                            text: moves[1].title.toUpperCase(),
+                            callback: () => this.teachMove(pokemon, move, 1),
+                        },
+                        {
+                            text: moves[2].title.toUpperCase(),
+                            callback: () => this.teachMove(pokemon, move, 2),
+                        },
+                        {
+                            text: moves[3].title.toUpperCase(),
+                            callback: () => this.teachMove(pokemon, move, 3),
+                        }],
+                });
+                this.gameStarter.menuGrapher.setActiveMenu("BattleFightList");
+            },
+        );
+        this.gameStarter.menuGrapher.setActiveMenu("GeneralText");
+    }
+
+    /**
+     * A function that gives the final dialog lines for resolving a move conflict as well as actually teaches
```

Comment:
Nit: check your doc comments for things that don't add to the sentence.

Example here: "A function that" isn't necessary; just say "Gives...".

Common other examples:
* "The thing being modified" -> "Thing being modified"
* "Whether or not something is true" -> "Whether something is true"